### PR TITLE
Modify - UI is_logged() refactored

### PR DESCRIPTION
**Pull request by rplevka**
This should address an issue: #2089

Note that by calling this method, the sessions `idle_timeout` is being reset as server treats the HTTP GET request as an activity from the users side, so one should reconsider using this method to test `idle_timeout` functionality.

### CODE REVIEW

File: robottelo/ui/login.py [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # -*- encoding: utf-8 -*-
 """Implements Login UI"""
-
```
Comment (elyezer) [inline robottelo/ui/login.py]: :+1:

File: robottelo/ui/login.py [hunk 2]
```diff
@@ -38,8 +38,25 @@ def logout(self):
         self.wait_for_ajax()
 
     def is_logged(self):
-        """Checks whether an user is logged"""
-        if self.wait_until_element(locators["login.gravatar"]):
+        """Checks whether an user is logged by validating the session cookie"""
+        cookies = self.browser.get_cookies()
```
Comment (elyezer) [inline robottelo/ui/login.py]: s/cookies = self.browser.get_cookies()/cookies = {}

File: robottelo/ui/login.py [hunk 3]
```diff
@@ -38,8 +38,25 @@ def logout(self):
...
+        wd_cookies = self.browser.get_cookies()
+        cookies = {}
```
Comment (elyezer) [inline robottelo/ui/login.py]: Drop this after changing as per last comment

File: robottelo/ui/login.py [hunk 4]
```diff
@@ -38,8 +38,25 @@ def logout(self):
...
+        # reformatting the cookies for requests.get request
+        for cookie in wd_cookies:
+            cookies[cookie['name']] = cookie['value']
+        # extract the root from the URL
+        url_root = self.browser.current_url.split('/')[2]
```
Comment (elyezer) [inline robottelo/ui/login.py]: Why not use get_server_url helper method here?

File: robottelo/ui/login.py [hunk 5]
```diff
@@ -38,8 +38,25 @@ def logout(self):
...
+
+        r = requests.get(
+            'https://'+url_root,
```
Comment (elyezer) [inline robottelo/ui/login.py]: get_server_url should give you the right url, without the need of prepending 'https://'

**Comment by elyezer** [comment]
ACK, pending comments

**Comment by elyezer** [comment]
This PR should be squash to only one commit.

File: robottelo/ui/login.py [hunk 6]
```diff
@@ -38,8 +39,24 @@ def logout(self):
...
 
     def is_logged(self):
-        """Checks whether an user is logged"""
-        if self.wait_until_element(locators["login.gravatar"]):
+        """Checks whether an user is logged by validating the session cookie"""
+        cookies = {}
+        wd_cookies = self.browser.get_cookies()
+        # reformatting the cookies for requests.get request
+        for cookie in wd_cookies:
+            cookies[cookie['name']] = cookie['value']
```
Comment (Ichimonji10) [inline robottelo/ui/login.py]: The above several lines of code can be compacted down to this:

```python
cookies = {
    cookie['name']: cookie['value']
    for cookie in self.browser.get_cookies()
}
```

Or, if we care about Python 2.6 compatibility:

```python
cookies = dict(
    (cookie['name'], cookie['value'])
    for cookie in self.browser.get_cookies()
)
```

File: robottelo/ui/login.py [hunk 7]
```diff
@@ -38,8 +39,24 @@ def logout(self):
...
+        # construct the link to the Dashboard web page
+        url_root = get_server_url() + "/dashboard"
```
Comment (Ichimonji10) [inline robottelo/ui/login.py]: `s/"/'/g`

File: robottelo/ui/login.py [hunk 8]
```diff
@@ -38,8 +39,24 @@ def logout(self):
...
+
+        r = requests.get(
```
Comment (Ichimonji10) [inline robottelo/ui/login.py]: No single-letter variable names, please. How about `response` instead?

File: robottelo/ui/login.py [hunk 9]
```diff
@@ -38,8 +39,24 @@ def logout(self):
...
+            allow_redirects=False,
+            cookies=cookies
+        )
+        if r.status_code == 302 and r.headers['location'].endswith('/login'):
+            return False
+        elif r.status_code == 200:
             return True
         else:
-            return False
+            r.raise_for_status()
```
Comment (Ichimonji10) [inline robottelo/ui/login.py]: How about this?

```python
response.raise_for_status()
if response.status_code == 302 and response.headers['location'].endswith('/login'):
    return False
elif response.status_code == 200:
    return True
else:
    # I have no idea. Be creative here?
```

**Comment by Ichimonji10** [comment]
ACK pending comments and a squash.

**Comment by rplevka** [comment]
All, comments addressed, commit ammended

File: robottelo/ui/login.py [hunk 10]
```diff
@@ -38,8 +39,20 @@ def logout(self):
...
+        url_root = get_server_url() + '/dashboard'
+
+        response = requests.get(
+            url_root,
+            verify=False,
+            allow_redirects=False,
+            cookies=cookies
+        )
+        response.raise_for_status()
+        return response.status_code != 302 \
```
Comment (elyezer) [inline robottelo/ui/login.py]: Using `\` for line continuation is not a good practice, please use `()` instead like:

```python
        return (
            response.status_code != 302 or
            not response.headers['location'].endswith('/login')
        )
```

or 

```python
        return (response.status_code != 302 or
                not response.headers['location'].endswith('/login'))
```

**Comment by elyezer** [comment]
ACK, pending comment.

**Comment by Ichimonji10** [comment]
Thank you for the hard work, @rplevka.

**Pull request merged**